cmd/zed/zed.d/statechange-led.sh

```diff
@@ -94,13 +98,44 @@ check_and_set_led()
 
 state_to_val()
 {
-	state="$1"
+	led="$1"
+	state="$2"
+
+	fault=""
+	locate=""
+
+	# Enumerate based on LED states.
 	case "$state" in
-		FAULTED|DEGRADED|UNAVAIL)
-			echo 1
-			;;
 		ONLINE)
-			echo 0
+			fault=0
+			locate=0
+			;;
+		OFFLINE)
+			fault=0
+			locate=1
+			;;
+		FAULTED|DEGRADED)
+			fault=1
+			locate=0
+			;;
+		REMOVED|UNAVAIL)
+			fault=1
+			locate=1
+			;;
+		*)
+			zed_log_msg "$0 state_to_val state='$state': LED mapping unknown"
```

Comment:
these were empty on purpose: no echo is an explicit "don't care", and there are many valid states not listed here, which my version preserved, but this breaks

(also, I find this horrendous readability-wise, but)

Reply:
My main goal with 138c8d24066bc228ccfa31a3d7a1d30629bdf2e8 is to make a clearer mapping between led status and vdev state. This is all a work in progress, which is why I'm leaving all commits in order to be able to track/backtrack until we can reach some agreement of what's best.

The logging is thought to be a fallback if a state is added to core zfs but not to the script, if it's logged sooner or later someone will discover it and assess whether it's relevant to react to or not.

The current list of states is what I found in the zpoolconcepts man page, if there are many more this might need to be done a bit differently.